### PR TITLE
feat: Group dependency updates

### PR DESCRIPTION
**Pull request by l0b0**
#### Motivation

Ensure that all CDK-related packages are updated together, for safety.

#### Checklist

- [ ] Tests updated (configuration change only)
- [ ] Docs updated (no docs necessary)
- [ ] Issue linked in Title (no related issue)

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -10,9 +10,14 @@ updates:
...
   open-pull-requests-limit: 10
   groups:
-    aws:
+    aws-sdk:
       patterns:
         - "@aws-sdk/*"
+    aws-cdk:
+      patterns:
+        - "aws-cdk*"
+        - "cdk8s*"
```
Comment (blacha) [inline .github/dependabot.yml:19]: this is too wide of a scope, with `@foo/*` we limit to a organisation with `cdk8s*` this includes anything prefixed eg `cdk8s-this-is-a-random-name`

Maybe one day cdk8s will switch to a org!

Comment (l0b0) [reply .github/dependabot.yml]: I've changed it.

Comment (l0b0) [reply .github/dependabot.yml]: & fixed it :grin

Comment (l0b0) [reply .github/dependabot.yml]: & simplified it :sweat_smile:

**Review by blacha — APPROVED** [review]

**Pull request merged**